apollo-federation/src/sources/connect/validation/mod.rs

```diff
@@ -581,6 +597,23 @@ impl Location {
             .map(|(line, column)| Location { line, column })?;
         Some(Range { start, end })
     }
+
+    fn from_abstract_type_node<T>(
```

Comment:
You could turn this into a top-level `unsupported_abstract_type` function and return the `Message` from up above. Then that'll unify how the error messages are formatted, too.